### PR TITLE
[hiredis] update to 1.0.0 (Fix #13256)

### LINKED ISSUE #13256: [hiredis] update to 1.0.0
Library name: hiredis

New version number: 1.0.0

Other information that may be useful (release notes, etc...)
https://github.com/redis/hiredis/blob/v1.0.0/CHANGELOG.md

### PR DESCRIPTION
**Pull request by klalumiere**
Updates the port `hiredis` to `v1.0.0`

- What does your PR fix? Fixes #13256

- Which triplets are supported/not supported? Have you updated the CI baseline?
Triplet support should be the same as before. I didn't touch to the CI baseline.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)? Yes

### CODE REVIEW

File: ports/hiredis/fix-feature-example.patch [hunk 1]
```diff
@@ -54,54 +54,13 @@ index dd3a313..8c69d3a 100644
...
 +TARGET_LINK_LIBRARIES(example hiredis ${LIBEVENT_LIBS})
-diff --git a/examples/example-libuv.c b/examples/example-libuv.c
-index a5462d4..9b7ca3e 100644
---- a/examples/example-libuv.c
-+++ b/examples/example-libuv.c
-@@ -33,7 +33,9 @@ void disconnectCallback(const redisAsyncContext *c, int status) {
- }
- 
- int main (int argc, char **argv) {
-+#ifndef _WIN32
```
Comment (klalumiere) [inline ports/hiredis/fix-feature-example.patch:72]: These issues have been fixed in v1.0.0 of https://github.com/redis/hiredis

File: ports/hiredis/fix-pthread.h-not-found-on-windows.patch [hunk 2]
```diff
@@ -1,16 +0,0 @@
-diff --git a/ssl.c b/ssl.c
```
Comment (klalumiere) [inline ports/hiredis/fix-pthread.h-not-found-on-windows.patch:1]: These issues have been fixed in v1.0.0 of https://github.com/redis/hiredis

**Comment by NancyLi1013** [comment]
Have you tested these features? Can they work fine?

**Comment by klalumiere** [comment]
> Have you tested these features? Can they work fine?

I was wondering what was microsoft/vcpkg's CI testing and what I needed to test on my side. I think perhaps I've read about that somewhere and forgotten.

Turns out I did run some test with our CI (which build the package on Ubuntu 20.04 and on Windows), with and without the feature `ssl` (disregard the SHA):

image

I'll try a run with the feature `example`, which is the only I didn't test.

**Comment by klalumiere** [comment]
`hiredis[example]` has the dependency `libevent`, which runs into some trouble both on Ubuntu 20.04 and Windows. This is unrelated to this pull request. There's already the issue #11680 which is open for the problem with `libevent` on Linux.

**Comment by NancyLi1013** [comment]
> I was wondering what was microsoft/vcpkg's CI testing and what I needed to test on my side. I think perhaps I've read about that somewhere and forgotten.
> 

vcpkg CI testing will test if these ports can build successfully on supported platforms or triplets. For those features (Except for default features) of ports, we need to test them on our machine manually. To make all features work fine, we will check this point to avoid new problems being involved in new commits.

**Comment by NancyLi1013** [comment]
> `hiredis[example]` has the dependency `libevent`, which runs into some trouble both on Ubuntu 20.04 and Windows. This is unrelated to this pull request. There's already the issue #11680 which is open for the problem with `libevent` on Linux.

`hiredis[example]` build failed in current master branch, but which is not related with the issue #11680. 

It failed with the following errors:

CMake Error at F:/vcpkg/installed/x86-windows/share/libevent/LibeventConfig.cmake:172 (message):
  Can not find any libraries for libevent.
Call Stack (most recent call first):
  F:/vcpkg/scripts/buildsystems/vcpkg.cmake:536 (_find_package)
  examples/CMakeLists.txt:22 (FIND_PACKAGE)

It is related with the usage of `libevent`. I will investigate it later.

**Comment by klalumiere** [comment]
> > `hiredis[example]` has the dependency `libevent`, which runs into some trouble both on Ubuntu 20.04 and Windows. This is unrelated to this pull request. There's already the issue #11680 which is open for the problem with `libevent` on Linux.
> 
> `hiredis[example]` build failed in current master branch, but which is not related with the issue #11680.

Interesting. I had the same error on our CI with my modifications, and I knew it was not directly related to #11680, but I still suspected a problem with the port `libevent` because of the warning [1]. It is a if the file `LibeventConfig.cmake` was broken.

[1]
```shell
CMake Warning at C:/jenkins/slaves/workspace/TestVcpkgPort/vcpkg/installed/x64-windows/share/libevent/LibeventConfig.cmake:65 (message):
  Your libevent library does not contain a core component!
Call Stack (most recent call first):
  C:/jenkins/slaves/workspace/TestVcpkgPort/vcpkg/installed/x64-windows/share/libevent/LibeventConfig.cmake:78 (message_if_needed)
  C:/jenkins/slaves/workspace/TestVcpkgPort/vcpkg/installed/x64-windows/share/libevent/LibeventConfig.cmake:140 (no_component_msg)
  C:/jenkins/slaves/workspace/TestVcpkgPort/vcpkg/installed/x64-windows/share/libevent/LibeventConfig.cmake:145 (find_event_lib)
  C:/jenkins/slaves/workspace/TestVcpkgPort/vcpkg/scripts/buildsystems/vcpkg.cmake:622 (_find_package)
  examples/CMakeLists.txt:22 (FIND_PACKAGE)


CMake Warning at C:/jenkins/slaves/workspace/TestVcpkgPort/vcpkg/installed/x64-windows/share/libevent/LibeventConfig.cmake:65 (message):
  Your libevent library does not contain a extra component!
Call Stack (most recent call first):
  C:/jenkins/slaves/workspace/TestVcpkgPort/vcpkg/installed/x64-windows/share/libevent/LibeventConfig.cmake:78 (message_if_needed)
  C:/jenkins/slaves/workspace/TestVcpkgPort/vcpkg/installed/x64-windows/share/libevent/LibeventConfig.cmake:140 (no_component_msg)
  C:/jenkins/slaves/workspace/TestVcpkgPort/vcpkg/installed/x64-windows/share/libevent/LibeventConfig.cmake:145 (find_event_lib)
  C:/jenkins/slaves/workspace/TestVcpkgPort/vcpkg/scripts/buildsystems/vcpkg.cmake:622 (_find_package)
  examples/CMakeLists.txt:22 (FIND_PACKAGE)
```

**Comment by klalumiere** [comment]
So the results of the checks following the `azp run` are scary. :fearful: Should I do anything? I've took a quick look at the output of the checks, but I'm not even sure that the problems they encountered are related to hiredis. :thinking:

**Comment by NancyLi1013** [comment]
The failures on CI tests are not related with this PR, caused by `getdns`.

**Comment by klalumiere** [comment]
The failures again seems unrelated to `hiredis`. I checked and the ports that failed do not depend (directly at least) on hiredis.

**Comment by NancyLi1013** [comment]
I have submitted a PR to fix `libevent `. I think it looks good now.

Thanks for your work @klalumiere.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by klalumiere** [comment]
Super thanks! :tada:

**Comment by strega-nil** [comment]
Thanks @klalumiere !

**Pull request merged**